src/main/scala/codacy/codesniffer/docsgen/parsers/WordPressCSDocsParser.scala

```diff
@@ -18,9 +18,29 @@ class WordPressCSDocsParser extends DocsParser {
     val sniffRegex(sniffType, patternName) = relativizedFilePath
     PatternIdParts("WordPress", sniffType, patternName)
   }
+
   override def descriptionWithDocs(rootDir: File,
-                                   patternIdParts: PatternIdParts): (Pattern.Description, Option[String]) = {
-    (descriptionFor(patternIdParts), None)
+                                   patternIdParts: PatternIdParts, patternFile: File): (Pattern.Description, Option[String]) = {
+    val description = descriptionFor(patternIdParts)
+    if(deprecated(patternFile)){
+      val newTitle = Pattern.Title(description.title.value.concat(" (Deprecated)"))
+      val newDescription = Pattern.Description(description.patternId, newTitle, description.description, description.timeToFix, description.parameters)
+      (newDescription, None)
+    } else {
+      (description, None)
+    }
+  }
+
+  private[this] def deprecated(patternFile: File): Boolean = {
```

Comment:
This seems like a really hardcore way to detect the deprecated. Can't we get this information at the same time we are parsing the pattern remaining information?

@prendi I believe you did this in a better way, am I mistaken?

Reply:
In this repo the information seems to be parsed from the file name only.
Maybe I'm mistaken, but I think @prendi did this for the parameters by filtering the ones that are deprecated when getting the result from the tool.
However in these cases what we want is to actually have some info saying the patterns are deprecated, and not filter them out.

Reply:
I had implemented something similar before but it was removed since it was no longer necessary.
Check this commit where I removed that logic: d27229fc3b056648564c65b5f2c0f91da489df02
I think the way I parsed it was simpler

Reply:
Yeah, it does look better. @rtfpessoa Does it look like a better solution to you?

Reply:
Then use his seems a bit better.